vivo-project/Vitro
# [VIVO-1528] Select an editor for input fields based on annotation

**[JIRA Issue](https://jira.duraspace.org/projects/VIVO)**: https://jira.duraspace.org/browse/VIVO-1528

# What does this pull request do?

It selects an editor based on the datatype annotation. If there is no such annotation, then the editor will be selected based an the range of the property to be edited.

# What's new?
WYSIWYG editor will be provided for data properties annotated with vitro:editing="HTML":
vivo:overview, vivo:teachingOverview, vivo:researchOverview, vivo:outreachOverview

# How should this be tested?

1. Edit a property having a range of integer, int, float, boolean, URI/URL or all types of date/time -> same behavior as before
2. Edit a property having a range of string or untyped and NOT having the annotation vitro:editing="HTML" -> input field should be provided (this is the new behavior)
3. Edit a property having the annotation vitro:editing="HTML" -> WYSIWYG editor should be provided

# Notice
1. You have to generate a firsttime start-up condition to apply the new annotations.
2. This pull request includes https://github.com/vivo-project/VIVO/pull/100 .
